server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSDelayOptionsCheckUtility.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.implicitDialects.cics.utility;
+
+        import org.antlr.v4.runtime.ParserRuleContext;
+
+        import org.eclipse.lsp.cobol.common.dialects.DialectProcessingContext;
+        import org.eclipse.lsp.cobol.common.error.ErrorSeverity;
+        import org.eclipse.lsp.cobol.common.error.SyntaxError;
+        import org.eclipse.lsp.cobol.implicitDialects.cics.CICSLexer;
+
+        import java.util.HashMap;
+        import java.util.List;
+        import java.util.Map;
+
+        import static org.eclipse.lsp.cobol.implicitDialects.cics.CICSParser.*;
+
+/** Checks CICS Delay rules for required and invalid options */
+public class CICSDelayOptionsCheckUtility extends CICSOptionsCheckBaseUtility {
+
+    public static final int RULE_INDEX = RULE_cics_delay;
+
+    private static final Map<Integer, ErrorSeverity> DUPLICATE_CHECK_OPTIONS =
+            new HashMap<Integer, ErrorSeverity>() {
+                {
+                    put(CICSLexer.DELAY, ErrorSeverity.ERROR);
+                    put(CICSLexer.INTERVAL, ErrorSeverity.ERROR);
+                    put(CICSLexer.TIME, ErrorSeverity.ERROR);
+                    put(CICSLexer.FOR, ErrorSeverity.ERROR);
+                    put(CICSLexer.HOURS, ErrorSeverity.ERROR);
+                    put(CICSLexer.MINUTES, ErrorSeverity.ERROR);
+                    put(CICSLexer.SECONDS, ErrorSeverity.ERROR);
+                    put(CICSLexer.MILLISECS, ErrorSeverity.ERROR);
+                    put(CICSLexer.UNTIL, ErrorSeverity.ERROR);
+                    put(CICSLexer.REQID, ErrorSeverity.ERROR);
+                }
+            };
+
+    public CICSDelayOptionsCheckUtility(
+            DialectProcessingContext context, List<SyntaxError> errors) {
+        super(context, errors, DUPLICATE_CHECK_OPTIONS);
+    }
+
+    /**
+     * Entrypoint to check CICS Delay rule options
+     *
+     * @param ctx ParserRuleContext subclass containing options
+     * @param <E> A subclass of ParserRuleContext
+     */
+    public <E extends ParserRuleContext> void checkOptions(E ctx) {
+        checkDuplicates(ctx);
```

Comment:
```suggestion
        checkMutuallyExclusiveOptions("INTERVAL, TIME, FOR or UNTIL", ctx.INTERVAL(), ctx.TIME(), ctx.FOR(), ctx.UNTIL());
        checkDuplicates(ctx);
```

These options are mutually exclusive to one another for DELAY.